pkg/scheduler/cache/cache.go

```diff
@@ -1536,9 +1537,12 @@ func (sc *SchedulerCache) UpdateJobStatus(job *schedulingapi.JobInfo, updatePG b
 		if err != nil {
 			return nil, err
 		}
+		sc.Mutex.Lock()
```

Comment:
I think there is no need to add lock? session --> UpdateJobStatus is still in the same goroutine, and there is no other goroutine will read or write this map

Reply:
I agree.

Reply:
According to the discussion, locks need to be added for all operations on the cache.

Reply:
Yes, l need to update podgroup to apiserver.